crates/solvers/src/domain/solution.rs

```diff
@@ -40,18 +67,53 @@ impl Trade {
 
     /// Returns the trade execution as an asset (token address and amount).
     pub fn executed(&self) -> eth::Asset {
-        match self.order.side {
-            order::Side::Buy => self.order.buy,
-            order::Side::Sell => self.order.sell,
+        let token = match self.order.side {
+            order::Side::Buy => self.order.buy.token,
+            order::Side::Sell => self.order.sell.token,
+        };
+
+        eth::Asset {
+            token,
+            amount: self.executed,
         }
     }
 }
 
+pub struct JitTrade {
+    pub order: order::JitOrder,
+    pub executed: U256,
+}
+
 /// A interaction included within an solution.
-pub struct Interaction {
+pub enum Interaction {
+    Liquidity(LiquidityInteraction),
+    Custom(CustomInteraction),
+}
+
+/// An interaction using input liquidity. This interaction will be encoded by
+/// the driver.
+pub struct LiquidityInteraction {
     pub liquidity: liquidity::Liquidity,
     // TODO: Currently there is not type-level guarantee that `input` and
     // output` are valid for the specified liquidity.
     pub input: eth::Asset,
     pub output: eth::Asset,
+    pub internalize: bool,
+}
+
+/// A custom interaction
```

Comment:
```suggestion
/// A custom interaction.
```

`CustomInteraction` has a doc comment saying "A custom interaction". This doesn't document anything, it only restates the name which we already see in the code. I would say: "an arbitrary interaction returned by the solver, which needs to be executed to fulfill the trade" or something similar to that.